File: packages-disabled/native-bridge/plugins/vision/ios/Sources/VisionPlugin.swift
import Foundation
import Capacitor

/**
 * A stub implementation of the Vision plugin for iOS.  Each method
 * currently calls reject() to indicate that the native OCR, face
 * detection, and barcode scanning are not yet implemented.  To
 * implement this plugin you would use the Apple Vision framework to
 * perform these recognitions and return the results to JavaScript.
 */
@objc(VisionPlugin)
public class VisionPlugin: CAPPlugin {
    @objc public func recognizeText(_ call: CAPPluginCall) {
        call.reject("Vision plugin recognizeText() is not implemented on iOS yet")
    }

    @objc public func detectFaces(_ call: CAPPluginCall) {
        call.reject("Vision plugin detectFaces() is not implemented on iOS yet")
    }

    @objc public func detectBarcodes(_ call: CAPPluginCall) {
        call.reject("Vision plugin detectBarcodes() is not implemented on iOS yet")
    }
}

File: packages-disabled/native-bridge/plugins/vision/android/src/main/java/com/cleanshare/vision/VisionPlugin.java
package com.cleanshare.vision;

import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/**
 * A stub implementation of the Vision plugin for Android.  It exposes
 * methods for text recognition, face detection, and barcode
 * detection, but currently rejects the call to indicate that the
 * native functionality has not been implemented.  You can extend
 * this class using Google ML Kit to return recogniser results.
 */
@CapacitorPlugin(name = "Vision")
public class VisionPlugin extends Plugin {
    @PluginMethod
    public void recognizeText(PluginCall call) {
        call.reject("Vision plugin recognizeText() is not implemented on Android yet");
    }

    @PluginMethod
    public void detectFaces(PluginCall call) {
        call.reject("Vision plugin detectFaces() is not implemented on Android yet");
    }

    @PluginMethod
    public void detectBarcodes(PluginCall call) {
        call.reject("Vision plugin detectBarcodes() is not implemented on Android yet");
    }
}